core/app/models/spree/product.rb

```diff
@@ -33,17 +33,17 @@ class Product < Spree::Base
     has_many :classifications, dependent: :delete_all, inverse_of: :product
     has_many :taxons, through: :classifications, before_remove: :remove_taxon
 
-    has_many :product_promotion_rules, class_name: 'Spree::ProductPromotionRule'
-    has_many :promotion_rules, through: :product_promotion_rules, class_name: 'Spree::PromotionRule'
+    has_many :product_promotion_rules, dependent: :destroy
```

Comment:
I think this PR should not introduce any unrelated changes, such as the removal of `class_name` in this case. Doing so, it would help reviewers job easier